README.rst
=============
Gmail Watcher
=============


A gmail notifier for Ubuntu with support for multiple accounts and instant
notifications. It let's you subscribe to selected folders in your gmail
account and notifies about new emails immediately. 


-------------
Requirements
-------------

Please check ``debian/control`` for runtime dependencies.


-------------
Installation
-------------

Install from repository and receive automatic updates::

    sudo add-apt-repository ppa:loneowais/gmailwatcher
    sudo apt-get update
    sudo apt-get install gmailwatcher

Use ``ppa:loneowais/gmailwatcher.dev`` for development builds.

OR

Install manually from tarball::

    sudo python setup.py install --record install.record --install-layout=deb

To remove::

     sudo rm $(cat install.record)


Notes
------

* "[Gmail]/All Mail" label must be present and visible via IMAP.
  Check the labels tab in your gmail settings to confirm.

* GmailWatcher will work on any linux distro as long as the required
  packages are installed. For example, it can work on OpenSUSE as Unity is
  packaged for it.


changelog
gmailwatcher (11.10) oneiric; urgency=low
    * Some changes the way unicode chars are handled
 -- Owais Lone <hello@owaislone.org>  Fri, 7 Oct 2011 16:49:04 +0530

gmailwatcher (11.09.14) oneiric; urgency=low
    * Some very minor UI changes
 -- Owais Lone <hello@owaislone.org>  Sun, 25 Sep 2011 19:27:20 +0530

gmailwatcher (11.09.13) oneiric; urgency=low
    * Fixes #847402
 -- Owais Lone <hello@owaislone.org>  Mon, 12 Sep 2011 3:00:20 +0530

gmailwatcher (11.09.12) oneiric; urgency=low
    * Added a pause switch
 -- Owais Lone <hello@owaislone.org>  Sun, 11 Sep 2011 15:04:20 +0530

gmailwatcher (11.09.11) oneiric; urgency=low
    * Some fixes in webkit UI (JS).
 -- Owais Lone <hello@owaislone.org>  Sun, 11 Sep 2011 13:23:45 +0530

gmailwatcher (11.08.26) oneiric; urgency=low
    * Temp fixes to saving/loading prefs [NEED TO REPLACE A LOT OF THIS CODE]
 -- Owais Lone <hello@owaislone.org>  Wed, 31 Aug 2011 22:13:5 +0530

gmailwatcher (11.08.25) oneiric; urgency=low
    * Experimenting with account switching
 -- Owais Lone <hello@owaislone.org>  Mon, 29 Aug 2011 1:20:14 +0530

gmailwatcher (11.08.24) oneiric; urgency=low
    * Started work on loading Gtk colors into webkit
    * Experimenting with account create form
 -- Owais Lone <hello@owaislone.org>  Sun, 21 Aug 2011 21:32:04 +0530

gmailwatcher (11.08.23) oneiric; urgency=low
    * Added custom style to scrollbars [unico engine]
    * Created proper directory structure for app icons
 -- Owais Lone <hello@owaislone.org>  Sun, 21 Aug 2011 00:01:49 +0530

gmailwatcher (11.08.22) oneiric; urgency=low
    * Fixed bug introduced in last commit
 -- Owais Lone <hello@owaislone.org>  Sat, 20 Aug 2011 18:22:01 +0530

gmailwatcher (11.08.21) oneiric; urgency=low
    * No startup notifications on autostart
    * Notifies about wrong credentials
    * New in app and launcher progress bars to show mail download progress.
 -- Owais Lone <hello@owaislone.org>  Sat, 20 Aug 2011 16:33:45 +0530

gmailwatcher (11.08.20) oneiric; urgency=low
    * Added scrollbars to show email download progress
    * Fixed bad strings
 -- Owais Lone <hello@owaislone.org>  Sat, 20 Aug 2011 02:38:03 +0530

gmailwatcher (11.08.19) oneiric; urgency=low
    * Changed autostart checkbox to gtk switch
 -- Owais Lone <hello@owaislone.org>  Fri, 19 Aug 2011 15:23:11 +0530

gmailwatcher (11.08.18) oneiric; urgency=low
    * Fixed a bug.
    * added temporary print statements 
 -- Owais Lone <hello@owaislone.org>  Tue, 16 Aug 2011 13:05:45 +0530

gmailwatcher (11.08.17~1) oneiric; urgency=low
    * Bumped dependency version
 -- Owais Lone <hello@owaislone.org>  Sat, 13 Aug 2011 21:05:00 +0530

gmailwatcher (11.08.16) oneiric; urgency=low
    * Improved PreferencesDialog
 -- Owais Lone <hello@owaislone.org>  Sat, 13 Aug 2011 13:01:34 +0530

gmailwatcher (11.08.15) oneiric; urgency=low
    * Added some more doc strings
    * Changed gir deps 
 -- Owais Lone <hello@owaislone.org>  Sat, 13 Aug 2011 2:49:00 +0530

gmailwatcher (11.08.14) oneiric; urgency=low
    * New Release
    * Fixed bug in preferences dialog
    * Now threads restart only when account changes
    * Added some doc strings
 -- Owais Lone <hello@owaislone.org>  Fri, 12 Aug 2011 16:03:09 +0530

gmailwatcher (11.08.13) oneiric; urgency=low
    * Changed dependency value to let it install on oneiric
 -- Owais Lone <hello@owaislone.org>  Fri, 12 Aug 2011 2:15:09 +0530

gmailwatcher (11.08.12) oneiric; urgency=low
    * Some theme changes
 -- Owais Lone <hello@owaislone.org>  Fri, 12 Aug 2011 00:15:09 +0530

gmailwatcher (11.08.11) oneiric; urgency=low
    * Added some notifications.
 -- Owais Lone <hello@owaislone.org>  Thu, 11 Aug 2011 20:14:45 +0530

gmailwatcher (11.08.10) oneiric; urgency=low
    * Now time checks are saved per label
 -- Owais Lone <hello@owaislone.org>  Thu, 11 Aug 2011 17:11:18 +0530

gmailwatcher (11.08.9) oneiric; urgency=low
    * Faster startup especially when already running AKA bring up unique instance
 -- Owais Lone <hello@owaislone.org>  Thu, 11 Aug 2011 12:15:11 +0530

gmailwatcher (11.08.8) oneiric; urgency=low
    * Disabled test emails
 -- Owais Lone <hello@owaislone.org>  Thu, 11 Aug 2011 11:28:11 +0530

gmailwatcher (11.08.7) oneiric; urgency=low
    * Worked on the theme a bit
 -- Owais Lone <hello@owaislone.org>  Thu, 11 Aug 2011 11:22:11 +0530



gmailwatcher/data/themes/dark/js/functions.js
var accounts = {}

var collapse_thread = function(thread) {
    messages = $(thread).children(':not(:last-child)').find('dt');
    $(messages).each(function(iter, elem) {
        $(elem).next().hide();
    })
}

var add_account = function(account) {
    var email = account.account;
    accounts[email] = {'folders': account.folders}

    folder_list = $('.folders[account='+email+']');
    if (folder_list.length == 0) {
        $('#foldersListTmpl').tmpl(account).appendTo('#sidebar');
        $('#emailsListTmpl').tmpl(account).appendTo('#content');
    }

    folders = []
    for (i in account.folders) {
        folders.push({
            'folder': account.folders[i],
            'account': email,
            'folder_id': i
          });
    };
    $('.folder[account='+email+']').remove();
    $('#folderTmpl').tmpl(folders).appendTo('.folders[account='+email+']');
    show_account(email);
};

var new_email = function(mail) {
    var account =  mail.account;
    var folder = mail.folder;
    var thread_id = mail.thread_id;
    mail.folder_id = accounts[account].folders.indexOf(folder);

    var email = $('#emailTmpl').tmpl(mail);
    thread = $('.email[thread_id='+thread_id+']');
    if (thread.length > 0) {
        email.find('.email-body').appendTo(thread);
    } else {
        mail_list = $('.emails[account='+account+']');
        email.prependTo(mail_list);
    };

    if (!$('.folder[account='+account+'][folder='+mail.folder_id+']').hasClass('selected')) {
        email.hide();
        $('.folder[account='+account+'][folder='+mail.folder_id+']').addClass('unseen');
        console.log($('.folder[account='+account+'][folder='+mail.folder_id+']'))
    }

    collapse_thread($('.email[thread_id='+thread_id+']'));
};

var show_account = function(account) {
    $('ul.folders').hide();
    $('ul.folders[account='+account+']').show();
    $('ul.folders[account='+account+']').children().first().click();
};

var test_data = function() {
    add_account({"folders": ["OdeskPS", "[GMAIL]/Important", "G+", "INBOX"], "account": "loneowais@gmail.com", "display_name": "Gmail"})


  new_email({"thread_id": "13762566744534478134",  "mail": [{"labels": [], "from": "\"GoDaddy.com\" <offers@godaddy.com>", "msg_id": "1376256674453447813", "thread_id": "1376256674453447813", "date": "4 Aug 2011 16:29:39 -0700", "starred": "", "system_labels": ["Important"], "subject": "Just Days left to SAVE 28%!"}], "account": "loneowais@gmail.com", "folder": "OdeskPS"});

    var i=0;

    while (i<1) {

new_email({"thread_id": "1376256674453447813",  "mail": [{"labels": [], "from": "\"GoDaddy.com\" <offers@godaddy.com>", "msg_id": "1376256674453447813", "thread_id": "1376256674453447813", "date": "4 Aug 2011 16:29:39 -0700", "starred": "", "system_labels": ["Important"], "subject": "Just Days left to SAVE 28%!"}], "account": "loneowais@gmail.com", "folder": "G+"});

new_email({"thread_id": "1376256674453447813",  "mail": [{"labels": [], "from": "\"GoDaddy.com\" <offers@godaddy.com>", "msg_id": "1376256674453447813", "thread_id": "1376256674453447813", "date": "4 Aug 2011 16:29:39 -0700", "starred": "", "system_labels": ["Important"], "subject": "Just Days left to SAVE 28%!"}], "account": "loneowais@gmail.com", "folder": "G+"});


new_email({"thread_id": "13762ds566744534478132",  "mail": [{"labels": [], "from": "\"GoDaddy.com\" <offers@godaddy.com>", "msg_id": "1376256674453447813", "thread_id": "1376256674453447813", "date": "4 Aug 2011 16:29:39 -0700", "starred": "", "system_labels": ["Important"], "subject": "Just Days left to SAVE 28%!"}], "account": "loneowais@gmail.com", "folder": "INBOX"});

    i = i+1;
    };
};

var theme = {};
theme.set_colors = function(color) {
  $('#sidebar, #sidebar *').css('background', color.bg)
  $('#sidebar, #sidebar *').css('border-color', color.border)
  $('#sidebar, #sidebar *').css('color', color.color)
};


$(document).ready(function() {

    $('.folder').live('click', function(event) {
        $('.folder').removeClass('selected');
        $('.email').hide();
        folder = $(this).attr('folder');
        account = $(this).attr('account');
        $('.email[account='+account+'][folder='+folder+']').show();
        $(this).addClass('selected');
        $(this).removeClass('unseen');
    });

    $('dt').live('click', function() {
        dd = $(this).next();
        if ($(dd).is(':hidden')) {
            $(dd).slideDown();
        }
        else {
            $(dd).slideUp();
        }
    });
   //test_data()
});


